lms/djangoapps/grades_report/grade_services.py

```diff
@@ -73,12 +74,45 @@ def by_section(self, section_block_id=None):
         and by assignment type data in that section.
 
         Returns:
-            1. section_at_breakdown: Object list by section, with assignment types info in that section.
+            1. grades: Object list by section.
             2. up_to_date_grade: Calculated value by student.
         """
         grades_data = GradeServices(self.course_id).get_grades_by_section(section_block_id)
         by_section_data = []
-        keys_to_delete = ['grade_breakdown', 'section_breakdown', 'section_filtered']
+        keys_to_delete = [
+            'grade_breakdown',
+            'section_breakdown',
+            'section_filtered',
+            'assignment_types'
+        ]
+        for item in grades_data['data']:
+            by_section_data.append(delete_unwanted_keys(item, keys_to_delete))
+        return by_section_data
+
+
+class ByAssignmentTypeGradeServices(object):
+    """
+    Class to generate by assignment type grade report.
+    """
+    def __init__(self, course_id=None):
+        self.course_id = course_id
+
+
+    def by_assignment_type(self, section_block_id=None):
```

Comment:
Thanks for splitting out this code @Squirrel18 , but since this method is an exact copy of `BySectionGradeServices.by_section`, the reports generated here will still be identical to those.

Can you please review the requirements doc and modify these reports to suit?